crates/nu-protocol/src/ast/expression.rs

```diff
@@ -325,26 +327,26 @@ impl Expression {
                 expr.replace_span(working_set, replaced, new_span);
             }
             Expr::Block(block_id) => {
-                let mut block = working_set.get_block(*block_id).clone();
+                let mut block = (**working_set.get_block(*block_id)).clone();
```

Comment:
That's maybe a little subtle in the syntax. Do we want to modify a clone of the contained value (or would we actually like to modify the underlying block at this block id if we could sanely?)

If it is the former, maybe worth pointing out the intention.
```suggestion
                // here we want to obtain a clone of the underlying `Block` to add to the `working_set` after mutation
                let mut block = (**working_set.get_block(*block_id)).clone();
```

Reply:
I wonder if I can make it look nicer with like `Block::clone` or something like that